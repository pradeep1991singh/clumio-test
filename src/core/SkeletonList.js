import React from "react";

const fakeArray = Array(12).fill(true);

const SkeletonList = () => {
  return (
    <div className="dark:bg-gray-900 p-10 grid gap-5 grid-cols-4">
      {fakeArray.map((item, i) => (
        <div key={i} className="dark:bg-gray-900">
          <div className="shadow rounded-md max-w-sm w-full mx-auto">
            <div className="bg-slate-500 h-40 w-30 rounded"></div>
            <div className="animate-pulse flex space-x-4">
              <div className="flex-1 space-y-6 py-1">
                <div className="space-y-3">
                  <div className="grid grid-cols-3 gap-3">
                    <div className="h-2 bg-slate-500 rounded col-span-2"></div>
                    <div className="h-2 bg-slate-500 rounded col-span-1"></div>
                  </div>
                  <div className="h-2 bg-slate-500 rounded"></div>
                </div>
              </div>
            </div>
          </div>
        </div>
      ))}
    </div>
  );
};

export default SkeletonList;
