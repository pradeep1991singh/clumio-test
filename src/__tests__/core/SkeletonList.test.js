import React from "react";
import { render } from "@testing-library/react";
import SkeletonList from "../../core/SkeletonList";

const defaultProps = {
  label: "label",
  name: "name",
  options: [
    { id: 1, name: "option1" },
    { id: 2, name: "option2" },
  ],
  onChange: jest.fn(),
};

describe("<SkeletonList />", () => {
  test("Snapshot", () => {
    const { asFragment } = render(<SkeletonList {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});