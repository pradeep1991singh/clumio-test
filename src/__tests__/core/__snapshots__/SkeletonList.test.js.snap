// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonList /> Snapshot 1`] = `
<DocumentFragment>
  <div
    class="dark:bg-gray-900 p-10 grid gap-5 grid-cols-4"
  >
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="dark:bg-gray-900"
    >
      <div
        class="shadow rounded-md max-w-sm w-full mx-auto"
      >
        <div
          class="bg-slate-500 h-40 w-30 rounded"
        />
        <div
          class="animate-pulse flex space-x-4"
        >
          <div
            class="flex-1 space-y-6 py-1"
          >
            <div
              class="space-y-3"
            >
              <div
                class="grid grid-cols-3 gap-3"
              >
                <div
                  class="h-2 bg-slate-500 rounded col-span-2"
                />
                <div
                  class="h-2 bg-slate-500 rounded col-span-1"
                />
              </div>
              <div
                class="h-2 bg-slate-500 rounded"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
